### PR TITLE
ipfs-cluster (alpha)

### PR DESCRIPTION
**Pull request by hsanjuan**
Features:

* Pin everywhere
* Unpin
* Any node can perform all actions, regardless of who is leader
* Check that it is actually pinned. retry failed pins.
* HTTP API
* CLI tool
* Server tool
* Docs
* 80% test cov
* gofmt happy
* golint happy
* go vet happy

See https://github.com/ipfs/ipfs-cluster/blob/devel/architecture.md for an overview of things and the issues for things missing which are coming down the pipe.

Happy holidays!

### CODE REVIEW

**Comment by jbenet** [comment]
This is awesome! \o/ i'm starting to review and play with ipfs-cluster

**Comment by 20zinnm** [comment]
I will take a look at this. Can we get a Dockerfile and/or kubernetes deployment?

File: README.md [hunk 1]
```diff
@@ -1,21 +1,139 @@
...
+
+After installing, you can run `ipfscluster --help` to display general description and options, or alternatively `ipfscluster help [cmd]` to display
+information about supported commands.
+
+In summary, it works as follows:
+
+```
+$ ipfscluster member ls                                                # list cluster members
+$ ipfscluster pin add Qma4Lid2T1F68E3Xa3CpE6vVJDLwxXLD8RfiB9g1Tmqp58   # pins a Cid in the cluster
+$ ipfscluster pin add Qma4Lid2T1F68E3Xa3CpE6vVJDLwxXLD8RfiB9g1Tmqp58   # unpins a Cid from the cluster
```
Comment (mateon1) [inline README.md]: There still is an issue here, the command `ipfscluster pin add` is repeated twice, for both pinning an unpinning. I believe this line should mention `ipfscluster pin rm` or similar.

**Comment by hsanjuan** [comment]
thanks everyone, I'm merging this now. I will add a roadmap to the project in a few mins. During this week i'll be working on addressing stuff from https://github.com/ipfs/ipfs-cluster/issues?q=is%3Aopen+is%3Aissue+milestone%3ABeta .

**Pull request merged**